examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,504 @@
+                                  system_messages.messages[con_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    feedback_agent = create_chat_agent("Feedback",
+                                       feedback_agent_config,
+                                       FEEDBACK_AGENT_SYSTEM_MESSAGE)
+    metaphor_search_agent = create_chat_agent("MetaphorSearch",
+                                              feedback_agent_config,
+                                              metaphor_search_agent_system_message)
+    logger.info("Pro, Con, feedback, and metaphor_search agents created.")
+
+    # Determine user's side and assign user_agent and ai_agent based on user selection
+    agents = {"pro": (pro_agent, con_agent, "Pro", "Con"), "con": (con_agent, pro_agent, "Con", "Pro")}
+    user_agent, ai_agent, user_side, ai_side = agents[side]
+    logger.info(
+        f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+        f"LLM Delegate: {llm_delegate}"
+    )
+
+    logger.info(f"\n{user_side} Agent ({topic_name}):\n")
+
+    # Determine if the debate is autonomous or the user input for one side
+    if llm_delegate:
+        logger.info("Autonomous Debate Selected")
+        interactive_setting = False
+    else:
+        logger.info("Manual Debate Selected with an AI Agent")
+        interactive_setting = True
+        user_input: str = Prompt.ask(
+            "Your argument (or type 'f' for feedback, 'done' to end):"
+        )
+        user_agent.llm = None  # User message without LLM completion
+        user_agent.user_message = user_input
+
+    # Set up langroid tasks and run the debate
+    user_task = Task(user_agent, interactive=interactive_setting, restart=False)
+    ai_task = Task(ai_agent, interactive=False, single_round=True)
+    user_task.add_sub_task(ai_task)
+    if not llm_delegate:
+        user_task.run(user_agent.user_message, turns=max_turns)
+    else:
+        user_task.run("get started", turns=max_turns)
+
+    # Determine the last agent based on turn count and alternation
+    # Note: user_agent and ai_agent are dynamically set based on the chosen user_side
+    last_agent = ai_agent if max_turns % 2 == 0 else user_agent
+
+    # Generate feedback summary and declare a winner using feedback agent
+
+    if not last_agent.message_history:
+        logger.warning("No agent message history found for the last agent")
+
+    feedback_agent_task = Task(feedback_agent, interactive=False)
+    formatted_history = parse_and_format_message_history(last_agent.message_history)
+    feedback_agent_task.run(formatted_history)  # Pass formatted history to the feedback agent
+
+    if metaphor_search:
+        metaphor_search_agent_task = Task(metaphor_search_agent, interactive=False)
+        metaphor_search_agent.enable_message(MetaphorSearchTool)
+        metaphor_search_agent.enable_message(DoneTool)
+        metaphor_search_agent_task.run("run the search")
```

Comment:
this agent gets stuck after:

```
- LLM -> search tool
-  agent_response -> results
- LLM -> final debate arguments with citations
```
It ends up working fine since langroid has a stalled loop check. But you can avoid that by having a `handle_message_fallback` method defined in a `MetaphorSearchAgent` subclassed from `ChatAgent`. You can use a variant of the definition here -- 
https://langroid.github.io/langroid/FAQ/#how-can-i-handle-an-llm-forgetting-to-generate-a-toolmessage

In your case you can just return `AgentDoneTool` to end the task, e.g.

```python
from langroid.agent.tools.orchestration import AgentDoneTool
def handle_message_fallback(
        self, msg: str | ChatDocument
    ) -> str | ChatDocument | None:
        """Handle scenario where LLM did not generate any Tool"""
        if isinstance(msg, ChatDocument) and msg.metadata.sender == Entity.LLM:
            return AgentDoneTool(content="done")
        return None

```

(Search for `handle_fallback_method` in `examples/` folder)

Reply:
fixed. thank you!